tests/lib/devicetree/api/src/main.c

```diff
@@ -1591,6 +1635,48 @@ static void test_child_nodes_list(void)
 	#undef TEST_FUNC
 }
 
+#undef DT_DRV_COMPAT
+#define DT_DRV_COMPAT vnd_child_bindings
+static void test_child_nodes_list_varg(void)
+{
+	#define TEST_FUNC(child, arg) { DT_PROP(child, val) + arg },
+	#define TEST_PARENT DT_PARENT(DT_NODELABEL(test_child_a))
+
+	struct vnd_child_binding {
+		int val;
+	};
+
+	struct vnd_child_binding vals[] = {
+		DT_FOREACH_CHILD_VARGS(TEST_PARENT, TEST_FUNC, 1)
+	};
+
+	struct vnd_child_binding vals_inst[] = {
+		DT_INST_FOREACH_CHILD_VARGS(0, TEST_FUNC, 1)
+	};
+
+	struct vnd_child_binding vals_status_okay[] = {
+		DT_FOREACH_CHILD_STATUS_OKAY_VARGS(TEST_PARENT, TEST_FUNC, 1)
+	};
+
+	zassert_equal(ARRAY_SIZE(vals), 3, "");
+	zassert_equal(ARRAY_SIZE(vals_inst), 3, "");
+	zassert_equal(ARRAY_SIZE(vals_status_okay), 2, "");
+
+	zassert_false(strlen(STRINGIFY(TEST_PARENT)) == 0, "");
```

Comment:
This assertion does not seem related to the test case. Is it debugging code? Can't it be removed?

Reply:
I followed the code pattern in `test_child_nodes_list`.

Reply:
It doesn't look reasonable for it to be in that function either; I've just sent a PR removing it. https://github.com/zephyrproject-rtos/zephyr/pull/36109. Thanks for pointing that out.

Reply:
I'll remove it from my code then and push the changes.